corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -27,37 +28,73 @@ hqDefine("data_dictionary/js/data_dictionary", [
+        self.groups = ko.observableArray();
+
+        self.init = function (groupData, changeSaveButton) {
+            for (let group of groupData) {
+                let groupObj = groupsViewModel(group.id, group.name, group.description, group.index, self.name, group.deprecated);
+                if (group.id) {
+                    groupObj.name.subscribe(changeSaveButton);
+                    groupObj.description.subscribe(changeSaveButton);
+                    groupObj.deprecated.subscribe(changeSaveButton);
+                }
+                groupObj.properties.subscribe(changeSaveButton);
 
-        self.init = function (groupDict, changeSaveButton) {
-            _.each(groupDict, function (properties, group) {
-                var groupObj = propertyListItem(group, '', true, group, self.name);
-                self.properties.push(groupObj);
-                properties.sort((a, b) => a.index - b.index);
-                _.each(properties, function (prop) {
+                for (let prop of group.properties) {
                     var propObj = propertyListItem(prop.name, prop.label, false, prop.group, self.name, prop.data_type,
                         prop.description, prop.allowed_values, prop.fhir_resource_prop_path, prop.deprecated,
-                        prop.removeFHIRResourcePropertyPath);
+                        prop.removeFHIRResourcePropertyPath, prop.index);
                     propObj.description.subscribe(changeSaveButton);
                     propObj.label.subscribe(changeSaveButton);
                     propObj.fhirResourcePropPath.subscribe(changeSaveButton);
                     propObj.dataType.subscribe(changeSaveButton);
                     propObj.deprecated.subscribe(changeSaveButton);
                     propObj.removeFHIRResourcePropertyPath.subscribe(changeSaveButton);
                     propObj.allowedValues.on('change', changeSaveButton);
-                    self.properties.push(propObj);
-                });
+                    groupObj.properties.push(propObj);
+                }
+
+                groupObj.properties.sort((a, b) => a.index - b.index);
+                self.groups.push(groupObj);
+            }
+
+            self.groups.sort((a, b) => {
+                return a.index && b.index && a.index - b.index;
             });
         };
 
         return self;
     };
 
+    var groupsViewModel = function (id, name, description, index, caseType, deprecated) {
+        var self = {};
+        self.id = id;
+        self.name = ko.observable(name);
+        self.description = ko.observable(description);
+        self.index = index;
+        self.caseType = caseType;
+        self.properties = ko.observableArray();
+        self.expanded = ko.observable(true);
+        self.toggleExpanded = () => self.expanded(!self.expanded());
+        self.deprecated = ko.observable(deprecated || false);
+        // Ensures that groups are not directly hidden on clicking the deprecated button
```

Comment:
can you explain this a bit?

Reply:
I added a variable that checks if the groups are deprecated when they are fetched from the DB. If a group is deprecated on page load from DB it is hidden from the UI. Otherwise If the user has clicked the button to deprecate a group it changes color to red and warning is shown, this variable helps in avoiding directly hiding a group on the click of deprecate button.

Reply:
So, we want a way to differentiate the ones that were already deprecated, and the ones that have been marked to be deprecated?
I would say, lets name these better to make it clear. How about?
```
self.toBeDeprecated = ko.observable(deprecated || false); // Or self.markedAsDeprecated
self.deprecated = deprecated
```

Reply:
Fixed. [1cadc80](https://github.com/dimagi/commcare-hq/pull/32878/commits/1cadc809a270298f41e0cfb49c16fab946f0fe08)